src/v/utils/process.cc

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "utils/process.h"
+
+#include "ssx/future-util.h"
+#include "vassert.h"
+#include "vlog.h"
+
+#include <seastar/core/loop.hh>
+#include <seastar/core/when_all.hh>
+#include <seastar/util/log.hh>
+
+#include <sstream>
+#include <string>
+
+static ss::logger proclog{"process"};
+
+namespace ssx {
+
+ss::future<>
+consume_input_stream(ss::input_stream<char> stream, bool is_stdout) {
```

Comment:
private helpers like this should be wrapped in the anonymous namespace to prevent symbol clashes.